lib/install/management/finder.go

```diff
@@ -46,67 +47,85 @@ func (d *Dispatcher) NewVCHFromID(id string) (*vm.VirtualMachine, error) {
 	ref, err := d.session.Finder.ObjectReference(d.ctx, *moref)
 	if err != nil {
 		if _, ok := err.(*find.NotFoundError); !ok {
-			err = errors.Errorf("Failed to query appliance (%s): %s", moref, err)
+			err = errors.Errorf("Failed to query appliance (%q): %s", moref, err)
 			return nil, err
 		}
 		log.Debugf("Appliance is not found")
 		return nil, nil
 	}
 	ovm, ok := ref.(*object.VirtualMachine)
 	if !ok {
-		log.Errorf("Failed to find VM %s, %s", moref, err)
+		log.Errorf("Failed to find VM %q, %s", moref, err)
 		return nil, err
 	}
 	vmm = vm.NewVirtualMachine(d.ctx, d.session, ovm.Reference())
 
 	// check if it's VCH
 	if ok, err = d.isVCH(vmm); err != nil {
-		log.Errorf("%s", err)
+		log.Error(err)
 		return nil, err
 	}
 	if !ok {
 		err = errors.Errorf("Not a VCH")
-		log.Errorf("%s", err)
+		log.Error(err)
 		return nil, err
 	}
 	return vmm, nil
 }
 
 func (d *Dispatcher) NewVCHFromComputePath(computePath string, name string, v *validate.Validator) (*vm.VirtualMachine, string, error) {
-	defer trace.End(trace.Begin(fmt.Sprintf("path %s, name %s", computePath, name)))
+	defer trace.End(trace.Begin(fmt.Sprintf("path %q, name %q", computePath, name)))
 
 	var err error
 
 	parent, err := v.ResourcePoolHelper(d.ctx, computePath)
 	if err != nil {
 		return nil, "", err
 	}
-	vchPath := fmt.Sprintf("%s/%s", parent.InventoryPath, name)
-	vchPool, err := v.ResourcePoolHelper(d.ctx, vchPath)
-	if err != nil {
-		log.Errorf("Failed to get VCH resource pool %s: %s", vchPath, err)
-		return nil, vchPath, err
+	d.vchPoolPath = path.Join(parent.InventoryPath, name)
+	var vchPool *object.ResourcePool
+	if d.isVC {
+		vapp, err := d.findVirtualApp(d.vchPoolPath)
+		if err != nil {
+			log.Errorf("Failed to get VCH virtual app %q, %s", d.vchPoolPath, err)
```

Comment:
looks like there are several of these, where comma should be colon in the format string.